Feature/0971 implement trash bin

VishalNehra opened the pull request:
<!-- 
Read this first:
To open a pull request read this file,
uncomment the corresponding lines and 
complete them.
-->

## Description

#### Issue tracker   
<!-- Fixes will automatically close the related issue -->
<!-- Fixes # -->
Fixes #971 
<!-- Addresses won't automatically close the related issue -->
<!-- Addresses # -->

#### Automatic tests
<!-- remember to do manual testing when making UI changes! -->
- [ ] Added test cases
  
#### Manual tests
- [x] Done  
  
<!-- If yes, -->
<!-- 
- Device:
- OS:
-->

#### Build tasks success  
<!-- run these! -->
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

<!-- If there are related PRs please add them here -->
<!--
#### Related PR  
Related to PR #
-->

VishalNehra committed:
971: Implement recycle bin

VishalNehra committed:
971: minor fix

VishalNehra committed:
Merge release/4.0

VishalNehra committed:
971: Implement trash bin

VishnuSanal reviewed (commented):
great work! basic features works as expected.

- a small bug I noticed was: delete a file `~/Picutes/Quotes/Quote.png` & delete/rename the directory `~/Picutes/Quotes/`. the file doesn't get restored now from the bin.

VishalNehra committed:
Merge upstream

VishalNehra committed:
971: Fix trash bin restore when parent file not present; fix directory icon in trash bin

TranceLove committed:
Fix DeleteTask failing delete SMB files

In most file manager implementations, including desktop ones, remote files are to be deleted directly even when put to recycle bin.

TranceLove committed:
Fix Codacy complaints

Also a condition inside DeleteTask, that delete SSH files should not trigger mediascanner as well.

TranceLove commented:
@VishalNehra Fixed test failures in 8dbfaec and Codacy complaints in bd77530.

There should be no option to allow user to put remote files (SMB/SSH/etc) to trash bin, and as I checked (Pixel 2 emulator running SDK 28), this had been covered in the UI level.

TranceLove committed:
Fix Codacy complaints on code complexity

eleius commented:
Is there a way to disable the trash bin? I couldn't find a setting. Not sure if using 0 for file # limit works?
